backend/routes/mcp_servers.py

```diff
@@ -122,51 +261,74 @@ async def sync_mcp_servers(db: AsyncSession):
                         "type": tool.get("type"),
                         "provider_id": tool.get("provider_id"),
                         "tool_host": tool.get("tool_host"),
-                        "parameters": [p.__dict__ for p in tool.get("parameters", [])]
-                    }
+                        "parameters": [p.__dict__ for p in tool.get("parameters", [])],
+                    },
                 }
-                
+
                 if tool.get("toolgroup_id") in existing_servers:
-                    print(f"Updating existing MCP server: {tool.get('toolgroup_id')}")
+                    logger.debug(
+                        f"Updating existing MCP server: {tool.get('toolgroup_id')}"
```

Comment:
Seems like mcp servers and tools are being conflated here. mcp servers corresponds to tool groups. Seems like we are setting tool attributes on mcp server entries and ending up with mcp server entries with attributes of the last tool in the tool group.